### PR TITLE
Make use of node_match & edge_match options & abstract add/remove of Q edge to separate method

### PR DESCRIPTION
**Pull request by kkafar**
I believe I've made it so implementing `is_matching_feasible` method is not necessary now, as all mappings returned from `GraphMatcher` (from `NetworkX`) should be feasible.

I'm still leaving possibility to overload that method in case above is not true or you need some stronger guarantees in your production than simply label (kind) & flag value match.

Also added method for adding/removing q-edge (which is now represented by separate node & four edges, not only two edges). You might need similar method for P-edge, but this is a good place to start.

### CODE REVIEW

**Review by MichalZelasko — APPROVED** [review]
Q hyperedge representation is now more understandable and logical. Thank you for implementing those changes.

File: src/graph.py [hunk 1]
```diff
@@ -43,7 +72,7 @@ def remove_edge_collection(self, edge_collection: Iterable[Edge]):
             self.remove_edge(edge.u, edge.v)
 
     def generate_subgraphs_isomorphic_with(self, other: 'Graph') -> Iterable[GraphMapping]:
-        gm = nx.isomorphism.GraphMatcher(self._graph, other.nx_graph)
+        gm = nx.isomorphism.GraphMatcher(self._graph, other.nx_graph, node_match=node_equality, edge_match=edge_equality)
         return gm.subgraph_isomorphisms_iter()
```
Comment (mrzelak) [inline src/graph.py:76]: nasty nit: this shouldn't be .subgraph_isomorphisms_iter, but .subgraph_monomorphisms_iter. See commit "graph.py: Change isomorphism to monomorphism" in my PR for further explanation

**Comment by mrzelak** [comment]
Added one nit which isn't relevant regarding those changes; also, in my fork I created hyperedge Q and P as a different node instead of edges, but speaking of code I think the approach presented here is more compact, and those two solutions can coexist; LGTM.

File: src/graph.py (hunk 1 above)
Comment (kkafar) [reply src/graph.py:76]: I see why we need monomorphism there, but let's fix that in dedicated PR

**Pull request merged**